tests/unit/processing/schema_migration/test_gather_collections.py

```diff
@@ -52,13 +72,15 @@ def test_with_auto_publish_false(self, schema_migrate_and_collections):
         schema_migrate, collections = schema_migrate_and_collections
         published_no_revision = collections["published"][0]
         private = collections["private"][0]
-        published_with_revision, revision = collections["revision"]
-        # get_collections is called twice
+        published_with_non_migration_revision, revision = collections["revision"]
+        published_with_migration_revision, migration_revision = collections["migration_revision"]
+        # get_collections is called thrice
```

Comment:
if `get_collections` is called thrice, then the side_effect, needs to be a list of len(3).
```suggestion
        # get_collections is called twice
```